### PR TITLE
Fixes locally rendered eye gaze

### PR DESCRIPTION
**Pull request by hyperlogic**
The MyAvatar eye gaze should follow the look at target when viewed from the mirror overlay.

This PR also adjusts when AnimVars for head and hand IK are set.  Now they are calculated and set *before* animation is performed.  This should reduce IK latency by one frame.

### CODE REVIEW

File: interface/src/avatar/SkeletonModel.cpp [hunk 1]
```diff
@@ -208,9 +220,16 @@ void SkeletonModel::updateRig(float deltaTime, glm::mat4 parentTransform) {
...
         head->setBaseYaw(glm::degrees(eulers.y));
         head->setBaseRoll(glm::degrees(-eulers.z));
-        _rig->updateEyeJoints(geometry.leftEyeJointIndex, geometry.rightEyeJointIndex,
-                              getTranslation(), getRotation(),
-                              head->getFinalOrientationInWorldFrame(), head->getCorrectedLookAtPosition());
+
+        Rig::EyeParameters eyeParams;
+        eyeParams.worldHeadOrientation = head->getFinalOrientationInWorldFrame();
+        eyeParams.eyeLookAt = head->getCorrectedLookAtPosition();
+        eyeParams.eyeSaccade = glm::vec3(0);
```
Comment (Atlante45) [inline interface/src/avatar/SkeletonModel.cpp]: Coding standard - `glm::vec3(0.0f);`
I always forget but I think you don't even need the `0.0f`, I think glm initialize to 0.0f by default.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b3734b5ddb3e44c6d5185abb86a3a0eb1992a456 available at: https://gist.github.com/aa3215096ec5575d59ed<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6205.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6205.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6ae5c540fdeb914ee6538de64f62056ba202ca74 available at: https://gist.github.com/aa3215096ec5575d59ed<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6205.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6205.dmg

File: libraries/animation/src/Rig.cpp [hunk 2]
```diff
@@ -990,6 +960,13 @@ void Rig::updateFromHeadParameters(const HeadParameters& params, float dt) {
     }
 }
 
+void Rig::updateFromEyeParameters(const EyeParameters& params, float dt) {
```
Comment (howard-stearns) [inline libraries/animation/src/Rig.cpp]: Introduced a new parameter, dt, which is then not used?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5e73af927237cc285337338e8efd4f5e354ffe52 available at: https://gist.github.com/aa3215096ec5575d59ed<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6205.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6205.dmg

**Pull request merged**